src/main/java/com/clarit/hs/service/items/PropertyCus.java

```diff
@@ -19,30 +42,61 @@ public List<Customer> getAll(String id){
     }
 
     @Override
-    public Customer add(String customerName){
-        return bookCustomer();
+    public Customer add(Customer customer){
+        return addCustomer(customer);
     }
 
+    @Override
+    public Customer update(Customer customer){
+        return updateCustomer(customer);
+    }
 
     @Override
     public List<Customer> get(String customerName){
+
         return itemRepositoryCus.findByName(customerName);
     }
 
+    @Override
+    public void delete(String name){
+        itemRepositoryCus.delete( itemRepositoryCus.deleteByName(name));
+    }
+    private List<Customer> getAllCustomers(){
+        return itemRepositoryCus.findAll();
+    }
 
+    private Customer addCustomer(Customer customer){
 
-   private List<Customer> getAllCustomers(){
-        return itemRepositoryCus.findAll();
-   }
-
-    private Customer bookCustomer(){
-        Customer customer = new Customer();
-        customer.setName("jerald");
-        customer.setEmail("jero@gmail.com");
-        customer.setAddress("senjai");
-        customer.setId("jero06");
-        customer.setPhone("9878877690");
         return itemRepositoryCus.save(customer);
+    }
 
+    private Customer updateCustomer(Customer customer){
+        return itemRepositoryCus.save(customer);
     }
+
+
+    @Autowired
+    ObjectMapper objectMapper;
+    @Override
+    public Customer patch(JsonPatch jsonPatch, String name) throws JsonPatchException, JsonProcessingException {
```

Comment:
do not throw JSON exceptions, service layer must throw custome service exceptions. handle them properly